SPEC.rdoc

```diff
@@ -202,30 +202,30 @@ This is an HTTP status. It must be an Integer greater than or equal to 100.
 
 === The Headers
 
-The headers must be a unfrozen +Hash+. The header keys must be +String+ objects. Special headers starting <tt>rack.</tt> are for communicating with the server, and must not be sent back to the client.
+The headers must be an unfrozen +Hash+. The header keys must be +String+ values. Special headers starting <tt>rack.</tt> are for communicating with the server, and must not be sent back to the client.
 
 * The headers must not contain a <tt>"status"</tt> key.
-* Header keys must conform to RFC7230 token specification, i.e. cannot contain non-printable ASCII, DQUOTE or <tt>(),/:;<=>?@[\]{}</tt>.
-* Header keys must not contain uppercase ASCII characters (A-Z).
-* Header values must be either a +String+ value, or an +Array+ of +String+ values, such that each +String+ value must not contain characters with an ASCII ordinal below 040 (32).
+* Header keys must conform to {RFC7230}[https://tools.ietf.org/html/rfc7230] token specification, i.e. cannot contain non-printable <tt>ASCII</tt>, <tt>DQUOTE</tt> or <tt>(),/:;<=>?@[\]{}</tt>.
```

Comment:
DQUOTE directly names the identifier used in RFC7230, so it makes sense to use `<tt>` for that.  I don't think it makes sense to do so for ASCII.

Reply:
```suggestion
* Header keys must conform to {RFC7230}[https://tools.ietf.org/html/rfc7230] token specification, i.e. cannot contain non-printable ASCII, <tt>DQUOTE</tt> or <tt>(),/:;<=>?@[\]{}</tt>.
```